### ISSUE TITLE
Ejs does not compile with node 6

### ISSUE THREAD

**Issue by Nighthawk14** [swanky-docs/swanky-core#28]
Trying to run swanky with node 6 throw an error :

```
node_modules/ejs/lib/ejs.js:237
  throw err;
  ^

TypeError: ejs:5
    3| <h3 id="__11__primary"><strong>1.1</strong> Primary</h3>
    4|
 >> 5| <%- render([{"template":"colour-drop","data":{"hex":"#21E779","rgb":"33 231 121","name":"Malachite"}},{"template":"colour-drop","data":{"hex":"#08B2B4","rgb":"200 255 100","name":"Deep Sky Blue"}},{"template":"colour-drop","data":{"hex":"#FFE11A","rgb":"255 225 26","name":"Lemon"}}]) %>
    6|
    7| <h3 id="__12__secondary"><strong>1.2</strong> Secondary</h3>
    8|

Path must be a string. Received [ { template: 'colour-drop',
    data: { hex: '#21E779', rgb: '33 231 121', name: 'Malachite' } },
  { template: 'colour-drop',
    data: { hex: '#08B2B4', rgb: '200 255 100', name: 'Deep Sky Blue' } },
  { template: 'colour-drop',
    data: { hex: '#FFE11A', rgb: '255 225 26', name: 'Lemon' } } ]
    at assertPath (path.js:7:11)
```

Didn't try to update ejs to see if that fix it :).

**Comment by rodleviton** [comment]
Will try and debug with Node 6 tonight.

**Comment by rodleviton** [comment]
@uglow @Nighthawk14 I just got the same issue on node 4. I have traced the issue to `getContent.js`. It seems that the two methods that are responsible for rendering the templates rely on arguments being available to determine what has been passed to them. In one of the latest refactors these methods were changed to the es6 fat arrow syntax which does not make arguments available.

```
 let render = () => { // arguments[0] undefined
 let render = function () { // arguments[0] string|array
```

I will fix this up and submit a new pull request.

**Comment by uglow** [comment]
Whoops! Sorry about that. I did a search for `this` but forgot to check for `arguments` Before doing the conversion.

**Comment by rodleviton** [comment]
@uglow. No problems. I did not know that with arrow functions arguments was not provided. Good learning for me.

**Comment by rodleviton** [comment]
@uglow it seems on [#L92](https://github.com/swanky-docs/swanky-core/blob/a77ada07be3d7e5338684e974d44aa29a99817f0/src/tasks/getContent.js#L92) that defining these two methods as let or const won't allow them to be passed to the ejs render method [#L150](https://github.com/swanky-docs/swanky-core/blob/a77ada07be3d7e5338684e974d44aa29a99817f0/src/tasks/getContent.js#L150) and causes the app to hang with no error output. I will have to revert the `render` and `code` method back to var unless you can think of a way around it.

**Comment by uglow** [comment]
Yes, EJS is ES5 only. I wasn't aware that this function was passed into EJS. So in this case  revert ES6 change.

Sent from my iPhone

On 22 Jul 2016, at 9:40 PM, Rod Leviton <notifications@github.com<mailto:notifications@github.com>> wrote:


@uglow<https://github.com/uglow> it seems on #L92<https://github.com/swanky-docs/swanky-core/blob/a77ada07be3d7e5338684e974d44aa29a99817f0/src/tasks/getContent.js#L92> that defining these two methods as let or const won't allow them to be passed to the ejs render method #L150<https://github.com/swanky-docs/swanky-core/blob/a77ada07be3d7e5338684e974d44aa29a99817f0/src/tasks/getContent.js#L150> and causes the app to hang with no error output. I will have to revert the render and code method back to var unless you can think of a way around it.

-

**Comment by rodleviton** [comment]
This should now be fixed in latest #33
